en/lessons/basics/basics.md

```diff
@@ -9,269 +9,197 @@ Getting started, basic data types, and basic operations.
+```
 
-Let's go ahead and give it a try now by typing in a few simple expressions:
+You can try out Haskell by typing in a few simple expressions:
 
-```elixir
-iex> 2+3
+```haskell
+ghci> 2 + 3
 5
-iex> 2+3 == 5
-true
-iex> String.length("The quick brown fox jumps over the lazy dog")
+ghci> 2 + 3 == 5
+True
+ghci> T.length "The quick brown fox jumps over the lazy dog"
 43
 ```
 
-Don't worry if you don't understand every expression yet, but we hope you get the idea.
+Perhaps you're able to figure out what some of these expressions mean.
 
 ## Basic Data Types
 
 ### Integers
 
-```elixir
-iex> 255
+Haskell has a bounded integer type called `Int`:
+
+```haskell
+ghci> 255 :: Int
 255
 ```
+__Note__: `::` specifies the type being used.
 
-Support for binary, octal, and hexadecimal numbers comes built in:
+Haskell also supports arbitrary precision integers with `Integer`:
 
-```elixir
-iex> 0b0110
-6
-iex> 0o644
-420
-iex> 0x1F
-31
+```haskell
+ghci> 123456789101112131415
+123456789101112131415
 ```
+__Note__: `ghci` automatically defaults to `Integer` here.
 
 ### Floats
 
-In Elixir, floating point numbers require a decimal after at least one digit; they have 64-bit double precision and support `e` for exponent values:
+Haskell supports `Double`, a double-precision floating-point number.
+At least one digit must come before the decimal point and Haskell supports
+scientific `e` notation.
 
-```elixir
-iex> 3.14
+```haskell
+ghci> 3.14
 3.14
-iex> .14
-** (SyntaxError) iex:2: syntax error before: '.'
-iex> 1.0e-10
-1.0e-10
-```
+ghci> .14
 
-### Booleans
-
-Elixir supports `true` and `false` as booleans; everything is truthy except for `false` and `nil`:
-
-```elixir
-iex> true
-true
-iex> false
-false
+<interactive>:2:1: error: parse error on input ‘.’
+ghci> 5e-10
+5.0e-10
+ghci> 2.7e10
+2.7e10
 ```
 
-### Atoms
-
-An atom is a constant whose name is its value.
-If you're familiar with Ruby, these are synonymous with Symbols:
-
-```elixir
-iex> :foo
-:foo
-iex> :foo == :bar
-false
-```
-
-The booleans `true` and `false` are also the atoms `:true` and `:false`, respectively.
-
-```elixir
-iex> is_atom(true)
-true
-iex> is_boolean(:true)
-true
-iex> :true === true
-true
-```
-
-Names of modules in Elixir are also atoms. `MyApp.MyModule` is a valid atom, even if no such module has been declared yet.
-
-```elixir
-iex> is_atom(MyApp.MyModule)
-true
-```
+### Booleans
 
-Atoms are also used to reference modules from Erlang libraries, including built in ones.
+Booleans in Haskell are represented with `True` and `False`:
 
-```elixir
-iex> :crypto.strong_rand_bytes 3
-<<23, 104, 108>>
+```haskell
+ghci> True
+True
+ghci> False
+False
 ```
 
-### Strings
+### Text
 
-Strings in Elixir are UTF-8 encoded and are wrapped in double quotes:
+Haskell supports unicode text. Strings are wrapped in double quotes:
 
-```elixir
-iex> "Hello"
+```haskell
+ghci> "Hello" :: T.Text
 "Hello"
-iex> "dziękuję"
-"dziękuję"
+ghci> "jalapeño" :: T.Text
+"jalape\241o"
```

Comment:
You may want to add that if the user wants the actual Unicode characters printed back (instead of the codepoints), they can use `putStrLn ("jalapeño" :: T.Text)`.

Reply:
I could go even further and print within the repl: 
>The constant type annotations may be confusing. A potential solution would be to use Data.Text.IO, but the extra import may be confusing; this solution would also enable for one to view unicode and escape codes working.

https://github.com/haskellfoundation/HaskellSchool/pull/7#discussion_r641925925

Do others agree with adding the cost of `Data.Text.IO`, and, furthermore, printing within the repl?